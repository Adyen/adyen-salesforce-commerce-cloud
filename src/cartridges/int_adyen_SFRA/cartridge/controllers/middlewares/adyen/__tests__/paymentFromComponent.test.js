/* eslint-disable global-require */
let paymentFromComponent;
let res;
let req;

beforeEach(() => {
  const { adyen } = require('../../index');
  paymentFromComponent = adyen.paymentFromComponent;
  jest.clearAllMocks();
  req = {
    form: {
      paymentMethod: 'method',
      data: {
        paymentMethod: {
          type: 'mocked_type'
        }
      }
    }
  };
  res = { redirect: jest.fn(), json: jest.fn() };
});

afterEach(() => {
  jest.resetModules();
});

describe('Payment from Component', () => {
  it('should cancel transaction', () => {
    const URLUtils = require('dw/web/URLUtils');

    req.form.data.cancelTransaction = true;
    req.form.data.merchantReference = 'mocked_merchantReference';
    req.form.data = JSON.stringify(req.form.data);
    paymentFromComponent(req, res, jest.fn());

    expect(URLUtils.url.mock.calls).toMatchSnapshot();
  });
  it('should return json response', () => {
    req.form.data = JSON.stringify(req.form.data);
    paymentFromComponent(req, res, jest.fn());
    expect(res.json.mock.calls).toMatchSnapshot();
  });
  it('should authorize express payment', () => {
    req.form.data.paymentMethod.paymentType = 'express';
    req.form.data = JSON.stringify(req.form.data);
    paymentFromComponent(req, res, jest.fn());
    expect(res.json.mock.calls).toMatchSnapshot();
  });
});
