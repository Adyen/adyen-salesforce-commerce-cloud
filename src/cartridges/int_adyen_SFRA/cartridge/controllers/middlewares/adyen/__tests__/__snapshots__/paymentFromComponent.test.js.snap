// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payment from Component should authorize express payment 1`] = `
[
  [
    {
      "isApplePay": true,
      "orderNo": "mocked_orderNo",
      "orderToken": "mocked_orderToken",
      "resultCode": "Authorised",
    },
  ],
]
`;

exports[`Payment from Component should cancel transaction 1`] = `
[
  [
    "Checkout-Begin",
    "stage",
    "placeOrder",
  ],
]
`;

exports[`Payment from Component should return json response 1`] = `
[
  [
    {
      "isApplePay": true,
      "orderNo": "mocked_orderNo",
      "orderToken": "mocked_orderToken",
      "resultCode": "Authorised",
    },
  ],
]
`;
